File: apk/app/src/main/java/com/example/momentra/ui/shell/group/living/LivingActiveTheme.kt
package com.example.momentra.ui.shell.group.living

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.momentra.ui.shell.group.GroupExperienceFamily
import com.example.momentra.ui.theme.PlusJakartaSans

/** Figma Flatmates 621:* / Co-living 629:* / Family Household 629:* / Custom Living 629:*. */
data class LivingActiveTheme(
    val bg: Color,
    val accent: Color,
    val accentLight: Color,
    val accentSoft: Color,
    val accentSolid: Color,
    val text: Color,
    val secondary: Color,
    val muted: Color,
    val card: Color,
    val border: Color,
    val darkText: Color,
    val typeLabel: String,
    val pulseTitle: String,
    val contributionsTitle: String,
    val budgetTitle: String,
    val insightsTitle: String,
    val healthLabel: String,
    val financeTitle: String,
    val hubHeroRes: Int,
    val heroEmoji: String,
    val participantRoles: List<String>,
    val participantSubtitle: String,
    val heroGradient: Brush,
    val pulseHeroGradient: Brush,
    val includesContribution: Boolean,
    val quickChips: List<Triple<String, String, LivingQuickAddKind>>,
    val statGradients: List<Brush>,
) {
    companion object {
        val Flatmates = LivingActiveTheme(
            bg = Color(0xFF14121B),
            accent = Color(0xFFE8744F),
            accentLight = Color(0xFFFF7A3D),
            accentSoft = Color(0x33E8744F),
            accentSolid = Color(0xFFEA580C),
            text = Color(0xFFE5E2E1),
            secondary = Color(0xFF9CA3AF),
            muted = Color(0xFFA8A19E),
            card = Color(0xFF1C1926),
            border = Color(0xFF2A2538),
            darkText = Color(0xFF14121B),
            typeLabel = "Flatmates",
            pulseTitle = "Household Pulse",
            contributionsTitle = "Member Contributions",
            budgetTitle = "Household Budget",
            insightsTitle = "Household Insights",
            healthLabel = "Household Health",
            financeTitle = "Group Finance",
            hubHeroRes = com.example.momentra.R.drawable.flatmates_hub_hero,
            heroEmoji = "🏠",
            participantRoles = listOf("Organizer", "Flatmate", "Guest"),
            participantSubtitle = "Add someone to the household roster",
            heroGradient = Brush.horizontalGradient(listOf(Color(0xFFFF7A3D), Color(0xFFE8744F))),
            pulseHeroGradient = Brush.linearGradient(
                colorStops = arrayOf(
                    0.25f to Color(0xFFFF7A3D),
                    0.75f to Color(0xFFEA580C).copy(alpha = 0.9f),
                ),
                start = Offset(40f, 0f),
                end = Offset(360f, 400f),
            ),
            includesContribution = true,
            quickChips = listOf(
                Triple("🏠", "Resident", LivingQuickAddKind.RESIDENT),
                Triple("💳", "Expense", LivingQuickAddKind.EXPENSE),
                Triple("🎁", "Contribute", LivingQuickAddKind.CONTRIBUTION),
                Triple("✅", "Task", LivingQuickAddKind.TASK),
            ),
            statGradients = listOf(
                Brush.horizontalGradient(listOf(Color(0xFFEA580C), Color(0xFFC2410C))),
                Brush.horizontalGradient(listOf(Color(0xFFE8744F), Color(0xFFEA580C))),
                Brush.horizontalGradient(listOf(Color(0xFFFF7A3D), Color(0xFFE8744F))),
                Brush.horizontalGradient(listOf(Color(0xFFFDBA74), Color(0xFFE8744F))),
            ),
        )

        // Figma 629:10541 hub — cyan #06B6D4
        val CoLiving = LivingActiveTheme(
            bg = Color(0xFF14121B),
            accent = Color(0xFF06B6D4),
            accentLight = Color(0xFF22D3EE),
            accentSoft = Color(0x3306B6D4),
            accentSolid = Color(0xFF0891B2),
            text = Color(0xFFE5E2E1),
            secondary = Color(0xFF9CA3AF),
            muted = Color(0xFFA8A19E),
            card = Color(0xFF1C1926),
            border = Color(0xFF2A2538),
            darkText = Color(0xFF14121B),
            typeLabel = "Co-living",
            pulseTitle = "Community Pulse",
            contributionsTitle = "Member Contributions",
            budgetTitle = "Community Budget",
            insightsTitle = "Community Insights",
            healthLabel = "Community Health",
            financeTitle = "Community Hub",
            hubHeroRes = com.example.momentra.R.drawable.coliving_hub_hero,
            heroEmoji = "🏢",
            participantRoles = listOf("Organizer", "Resident", "Guest"),
            participantSubtitle = "Invite people sharing this co-living space",
            heroGradient = Brush.horizontalGradient(listOf(Color(0xFF068CA6), Color(0xFF043744))),
            pulseHeroGradient = Brush.linearGradient(
                colorStops = arrayOf(
                    0.25f to Color(0xFF22D3EE),
                    0.75f to Color(0xFF0E7490).copy(alpha = 0.9f),
                ),
                start = Offset(40f, 0f),
                end = Offset(360f, 400f),
            ),
            includesContribution = true,
            quickChips = listOf(
                Triple("🏠", "Resident", LivingQuickAddKind.RESIDENT),
                Triple("💳", "Expense", LivingQuickAddKind.EXPENSE),
                Triple("🎁", "Contribute", LivingQuickAddKind.CONTRIBUTION),
                Triple("🔧", "Maintain", LivingQuickAddKind.MAINTENANCE),
            ),
            statGradients = listOf(
                Brush.horizontalGradient(listOf(Color(0xFF0E7490), Color(0xFF155E75))),
                Brush.horizontalGradient(listOf(Color(0xFF06B6D4), Color(0xFF0891B2))),
                Brush.horizontalGradient(listOf(Color(0xFF22D3EE), Color(0xFF06B6D4))),
                Brush.horizontalGradient(listOf(Color(0xFF67E8F9), Color(0xFF06B6D4))),
            ),
        )

        // Figma 629:16126 hub — amber #F59E0B
        val FamilyHousehold = LivingActiveTheme(
            bg = Color(0xFF14121B),
            accent = Color(0xFFF59E0B),
            accentLight = Color(0xFFFBBF24),
            accentSoft = Color(0x33F59E0B),
            accentSolid = Color(0xFFD97706),
            text = Color(0xFFE5E2E1),
            secondary = Color(0xFF9CA3AF),
            muted = Color(0xFFA8A19E),
            card = Color(0xFF1C1926),
            border = Color(0xFF2A2538),
            darkText = Color(0xFF14121B),
            typeLabel = "Family Household",
            pulseTitle = "Family Pulse",
            contributionsTitle = "Household Members",
            budgetTitle = "Family Budget",
            insightsTitle = "Family Insights",
            healthLabel = "Family Health",
            financeTitle = "Group Finance",
            hubHeroRes = com.example.momentra.R.drawable.family_household_hub_hero,
            heroEmoji = "👨‍👩‍👧",
            participantRoles = listOf("Organizer", "Family", "Guest"),
            participantSubtitle = "Add a family member to the household",
            heroGradient = Brush.horizontalGradient(listOf(Color(0xFFF59E0B), Color(0xFF78350F))),
            pulseHeroGradient = Brush.linearGradient(
                colorStops = arrayOf(
                    0.25f to Color(0xFFFBBF24),
                    0.75f to Color(0xFFB45309).copy(alpha = 0.9f),
                ),
                start = Offset(40f, 0f),
                end = Offset(360f, 400f),
            ),
            includesContribution = false,
            quickChips = listOf(
                Triple("🏠", "Resident", LivingQuickAddKind.RESIDENT),
                Triple("💳", "Expense", LivingQuickAddKind.EXPENSE),
                Triple("✅", "Task", LivingQuickAddKind.TASK),
                Triple("📷", "Photos", LivingQuickAddKind.MEMORY),
            ),
            statGradients = listOf(
                Brush.horizontalGradient(listOf(Color(0xFFB45309), Color(0xFF92400E))),
                Brush.horizontalGradient(listOf(Color(0xFFF59E0B), Color(0xFFD97706))),
                Brush.horizontalGradient(listOf(Color(0xFFFBBF24), Color(0xFFF59E0B))),
                Brush.horizontalGradient(listOf(Color(0xFFFCD34D), Color(0xFFF59E0B))),
            ),
        )

        // Figma 629:15586 hub — emerald #10B981
        val CustomLiving = LivingActiveTheme(
            bg = Color(0xFF14121B),
            accent = Color(0xFF10B981),
            accentLight = Color(0xFF34D399),
            accentSoft = Color(0x3310B981),
            accentSolid = Color(0xFF059669),
            text = Color(0xFFE5E2E1),
            secondary = Color(0xFF9CA3AF),
            muted = Color(0xFFA8A19E),
            card = Color(0xFF1C1926),
            border = Color(0xFF2A2538),
            darkText = Color(0xFF14121B),
            typeLabel = "Custom Living",
            pulseTitle = "Living Pulse",
            contributionsTitle = "Household Members",
            budgetTitle = "Living Budget",
            insightsTitle = "Living Insights",
            healthLabel = "Living Health",
            financeTitle = "Property Hub",
            hubHeroRes = com.example.momentra.R.drawable.custom_living_hub_hero,
            heroEmoji = "✨",
            participantRoles = listOf("Organizer", "Resident", "Member"),
            participantSubtitle = "Add someone to this living arrangement",
            heroGradient = Brush.horizontalGradient(listOf(Color(0xFF10B981), Color(0xFF0F766E))),
            pulseHeroGradient = Brush.linearGradient(
                colorStops = arrayOf(
                    0.25f to Color(0xFF34D399),
                    0.75f to Color(0xFF047857).copy(alpha = 0.9f),
                ),
                start = Offset(40f, 0f),
                end = Offset(360f, 400f),
            ),
            includesContribution = false,
            quickChips = listOf(
                Triple("🏠", "Resident", LivingQuickAddKind.RESIDENT),
                Triple("💳", "Expense", LivingQuickAddKind.EXPENSE),
                Triple("✅", "Task", LivingQuickAddKind.TASK),
                Triple("📷", "Photos", LivingQuickAddKind.MEMORY),
            ),
            statGradients = listOf(
                Brush.horizontalGradient(listOf(Color(0xFF047857), Color(0xFF065F46))),
                Brush.horizontalGradient(listOf(Color(0xFF10B981), Color(0xFF059669))),
                Brush.horizontalGradient(listOf(Color(0xFF34D399), Color(0xFF10B981))),
                Brush.horizontalGradient(listOf(Color(0xFF6EE7B7), Color(0xFF10B981))),
            ),
        )

        fun forFamily(family: GroupExperienceFamily): LivingActiveTheme =
            when (family) {
                GroupExperienceFamily.FAMILY_HOUSEHOLD -> FamilyHousehold
                GroupExperienceFamily.CO_LIVING -> CoLiving
                GroupExperienceFamily.CUSTOM_LIVING -> CustomLiving
                else -> Flatmates
            }
    }
}

enum class LivingQuickAddKind {
    RESIDENT,
    EXPENSE,
    CONTRIBUTION,
    TASK,
    RULE,
    ASSET,
    MAINTENANCE,
    UPDATE,
    POLL,
    MEMORY,
}

fun LivingQuickAddKind.label(): String = when (this) {
    LivingQuickAddKind.RESIDENT -> "Resident"
    LivingQuickAddKind.EXPENSE -> "Expense"
    LivingQuickAddKind.CONTRIBUTION -> "Contribution"
    LivingQuickAddKind.TASK -> "Task"
    LivingQuickAddKind.RULE -> "House Rule"
    LivingQuickAddKind.ASSET -> "Asset"
    LivingQuickAddKind.MAINTENANCE -> "Maintenance"
    LivingQuickAddKind.UPDATE -> "Update"
    LivingQuickAddKind.POLL -> "Poll"
    LivingQuickAddKind.MEMORY -> "Memory"
}

fun LivingQuickAddKind.emoji(): String = when (this) {
    LivingQuickAddKind.RESIDENT -> "🏠"
    LivingQuickAddKind.EXPENSE -> "💳"
    LivingQuickAddKind.CONTRIBUTION -> "🎁"
    LivingQuickAddKind.TASK -> "✅"
    LivingQuickAddKind.RULE -> "📋"
    LivingQuickAddKind.ASSET -> "📦"
    LivingQuickAddKind.MAINTENANCE -> "🔧"
    LivingQuickAddKind.UPDATE -> "📢"
    LivingQuickAddKind.POLL -> "📊"
    LivingQuickAddKind.MEMORY -> "📷"
}

fun LivingQuickAddKind.isLive(): Boolean = this != LivingQuickAddKind.RULE

fun livingHubTiles(theme: LivingActiveTheme): List<LivingQuickAddKind> {
    val tiles = mutableListOf(
        LivingQuickAddKind.RESIDENT,
        LivingQuickAddKind.EXPENSE,
    )
    if (theme.includesContribution) tiles.add(LivingQuickAddKind.CONTRIBUTION)
    tiles.add(LivingQuickAddKind.TASK)
    if (theme.includesContribution) {
        tiles.add(LivingQuickAddKind.RULE)
        tiles.add(LivingQuickAddKind.ASSET)
        tiles.add(LivingQuickAddKind.MAINTENANCE)
        tiles.add(LivingQuickAddKind.UPDATE)
        tiles.add(LivingQuickAddKind.POLL)
        tiles.add(LivingQuickAddKind.MEMORY)
    } else {
        // Family / Custom: Resident, Expense, Task, Asset, Maintenance, Update, Rule, Poll, Memory
        tiles.add(LivingQuickAddKind.ASSET)
        tiles.add(LivingQuickAddKind.MAINTENANCE)
        tiles.add(LivingQuickAddKind.UPDATE)
        tiles.add(LivingQuickAddKind.RULE)
        tiles.add(LivingQuickAddKind.POLL)
        tiles.add(LivingQuickAddKind.MEMORY)
    }
    return tiles
}

@Composable
fun LivingSectionCard(
    theme: LivingActiveTheme,
    title: String,
    modifier: Modifier = Modifier,
    trailing: (@Composable () -> Unit)? = null,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(theme.card)
            .border(1.dp, theme.border, RoundedCornerShape(20.dp))
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(title, color = theme.text, fontSize = 18.sp, fontWeight = FontWeight.ExtraBold, fontFamily = PlusJakartaSans)
            trailing?.invoke()
        }
        content()
    }
}

@Composable
fun LivingEmptyBlock(theme: LivingActiveTheme, message: String, detail: String) {
    Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
        Text(message, color = theme.text, fontSize = 13.sp, fontWeight = FontWeight.SemiBold, fontFamily = PlusJakartaSans)
        Text(detail, color = theme.secondary, fontSize = 12.sp, fontFamily = PlusJakartaSans)
    }
}

@Composable
fun LivingStatCard(label: String, value: String, gradient: Brush, modifier: Modifier = Modifier) {
    Column(
        modifier = modifier
            .clip(RoundedCornerShape(16.dp))
            .background(gradient)
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Text(label, color = Color.White.copy(alpha = 0.95f), fontSize = 10.sp, fontWeight = FontWeight.SemiBold, fontFamily = PlusJakartaSans)
        Text(value, color = Color.White.copy(alpha = 0.95f), fontSize = 22.sp, fontWeight = FontWeight.Bold, fontFamily = PlusJakartaSans)
    }
}

@Composable
fun LivingEmojiChip(
    theme: LivingActiveTheme,
    label: String,
    emoji: String,
    enabled: Boolean = true,
    onClick: () -> Unit = {},
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier.then(if (enabled) Modifier.clickable(onClick = onClick) else Modifier),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Box(
            modifier = Modifier
                .size(56.dp)
                .clip(RoundedCornerShape(16.dp))
                .background(theme.card)
                .border(1.dp, theme.border, RoundedCornerShape(16.dp)),
            contentAlignment = Alignment.Center,
        ) {
            Text(emoji, fontSize = 22.sp)
        }
        Text(label, color = theme.text, fontSize = 11.sp, fontWeight = FontWeight.SemiBold, fontFamily = PlusJakartaSans)
    }
}

@Composable
fun LivingCrewRow(
    theme: LivingActiveTheme,
    name: String,
    role: String,
    amountLabel: String,
    featured: Boolean = false,
) {
    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
        Box(
            modifier = Modifier
                .size(40.dp)
                .clip(CircleShape)
                .background(theme.accentSoft),
            contentAlignment = Alignment.Center,
        ) {
            Text(name.take(1).uppercase(), color = theme.accentLight, fontWeight = FontWeight.Bold, fontFamily = PlusJakartaSans)
        }
        Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(
                "$name ($role)${if (featured) " ★" else ""}",
                color = theme.text,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                fontFamily = PlusJakartaSans,
            )
            Text(if (featured) "Top contributor" else "Resident", color = theme.muted, fontSize = 11.sp, fontFamily = PlusJakartaSans)
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(6.dp)
                    .clip(RoundedCornerShape(999.dp))
                    .background(theme.accentSoft),
            )
        }
        Text(amountLabel, color = theme.accentLight, fontSize = 12.sp, fontWeight = FontWeight.Bold, fontFamily = PlusJakartaSans)
    }
}

File: momentra/momentra/Shell/GroupActive/Living/LivingActiveTheme.swift
import SwiftUI

/// Shared Living active tokens — Flatmates / Family Household / Co-living / Custom Living (G09–G12).
struct LivingActiveTheme {
    let bg: Color
    let accent: Color
    let accentSolid: Color
    let accentLight: Color
    let accentSoft: Color
    let text: Color
    let secondary: Color
    let muted: Color
    let card: Color
    let border: Color
    let darkText: Color
    let peachChip: Color
    let tealChip: Color
    let purpleChip: Color
    let typeLabel: String
    let pulseTitle: String
    let contributionsTitle: String
    let budgetTitle: String
    let financeTitle: String
    let hubHeroAssetName: String
    let heroEmoji: String
    let participantRoles: [String]
    let participantSubtitle: String
    let heroGradientColors: [Color]
    let pulseHeroGradientColors: [Color]
    let statGradients: [[Color]]
    let includesContribution: Bool
    let quickChips: [(emoji: String, label: String, kind: LivingQuickAddKind)]

    var heroGradient: LinearGradient {
        LinearGradient(colors: heroGradientColors, startPoint: .leading, endPoint: .trailing)
    }

    var pulseHeroGradient: LinearGradient {
        LinearGradient(
            colors: pulseHeroGradientColors,
            startPoint: UnitPoint(x: 0.1, y: 0),
            endPoint: UnitPoint(x: 0.9, y: 1)
        )
    }

    var sectionRadius: CGFloat { 20 }

    static let flatmates = LivingActiveTheme(
        bg: Color(hex: "#14121B"),
        accent: Color(hex: "#E8744F"),
        accentSolid: Color(hex: "#FF7A3D"),
        accentLight: Color(hex: "#FF7A3D"),
        accentSoft: Color(hex: "#E8744F").opacity(0.2),
        text: Color(hex: "#E5E2E1"),
        secondary: Color(hex: "#9CA3AF"),
        muted: Color(hex: "#9CA3AF"),
        card: Color(hex: "#1C1926"),
        border: Color(hex: "#2A2538"),
        darkText: Color(hex: "#14121B"),
        peachChip: Color(hex: "#FBBF24"),
        tealChip: Color(hex: "#14B8A6"),
        purpleChip: Color(hex: "#A855F7"),
        typeLabel: "Flatmates",
        pulseTitle: "Flatmates Pulse",
        contributionsTitle: "Member Contributions",
        budgetTitle: "Household Budget",
        financeTitle: "Group Finance",
        hubHeroAssetName: "FlatmatesHubHero",
        heroEmoji: "🏠",
        participantRoles: ["Organizer", "Resident", "Guest"],
        participantSubtitle: "Invite people living in this household",
        heroGradientColors: [Color(hex: "#FF7A3D"), Color(hex: "#E8744F")],
        pulseHeroGradientColors: [Color(hex: "#FDBA74"), Color(hex: "#EA580C").opacity(0.9)],
        statGradients: [
            [Color(hex: "#C2410C"), Color(hex: "#9A3412")],
            [Color(hex: "#E8744F"), Color(hex: "#EA580C")],
            [Color(hex: "#7C3AED"), Color(hex: "#5B21B6")],
            [Color(hex: "#FF7A3D"), Color(hex: "#C2410C")],
        ],
        includesContribution: true,
        quickChips: [
            ("👤", "Resident", .resident),
            ("💳", "Expense", .expense),
            ("✅", "Task", .task),
            ("📷", "Memory", .memory),
        ]
    )

    static let coLiving = LivingActiveTheme(
        bg: Color(hex: "#14121B"),
        // Figma 629:10541 hub — cyan #06B6D4
        accent: Color(hex: "#06B6D4"),
        accentSolid: Color(hex: "#06B6D4"),
        accentLight: Color(hex: "#22D3EE"),
        accentSoft: Color(hex: "#06B6D4").opacity(0.2),
        text: Color(hex: "#E5E2E1"),
        secondary: Color(hex: "#9CA3AF"),
        muted: Color(hex: "#9CA3AF"),
        card: Color(hex: "#1C1926"),
        border: Color(hex: "#2A2538"),
        darkText: Color(hex: "#14121B"),
        peachChip: Color(hex: "#FBBF24"),
        tealChip: Color(hex: "#14B8A6"),
        purpleChip: Color(hex: "#A855F7"),
        typeLabel: "Co-living",
        pulseTitle: "Co-living Pulse",
        contributionsTitle: "Member Contributions",
        budgetTitle: "Community Budget",
        financeTitle: "Community Hub",
        hubHeroAssetName: "ColivingHubHero",
        heroEmoji: "🏘️",
        participantRoles: ["Organizer", "Resident", "Member"],
        participantSubtitle: "Invite people sharing this co-living space",
        heroGradientColors: [Color(hex: "#068CA6"), Color(hex: "#043744")],
        pulseHeroGradientColors: [Color(hex: "#22D3EE"), Color(hex: "#0E7490").opacity(0.9)],
        statGradients: [
            [Color(hex: "#0E7490"), Color(hex: "#155E75")],
            [Color(hex: "#06B6D4"), Color(hex: "#0891B2")],
            [Color(hex: "#7C3AED"), Color(hex: "#5B21B6")],
            [Color(hex: "#22D3EE"), Color(hex: "#0E7490")],
        ],
        includesContribution: true,
        quickChips: [
            ("👤", "Resident", .resident),
            ("💳", "Expense", .expense),
            ("✅", "Task", .task),
            ("📷", "Memory", .memory),
        ]
    )

    static let familyHousehold = LivingActiveTheme(
        bg: Color(hex: "#14121B"),
        // Figma 629:16126 hub — amber #F59E0B
        accent: Color(hex: "#F59E0B"),
        accentSolid: Color(hex: "#F59E0B"),
        accentLight: Color(hex: "#FBBF24"),
        accentSoft: Color(hex: "#F59E0B").opacity(0.2),
        text: Color(hex: "#E5E2E1"),
        secondary: Color(hex: "#9CA3AF"),
        muted: Color(hex: "#9CA3AF"),
        card: Color(hex: "#1C1926"),
        border: Color(hex: "#2A2538"),
        darkText: Color(hex: "#14121B"),
        peachChip: Color(hex: "#FBBF24"),
        tealChip: Color(hex: "#14B8A6"),
        purpleChip: Color(hex: "#A855F7"),
        typeLabel: "Family Household",
        pulseTitle: "Family Pulse",
        contributionsTitle: "Member Contributions",
        budgetTitle: "Household Budget",
        financeTitle: "Group Finance",
        hubHeroAssetName: "FamilyHouseholdHubHero",
        heroEmoji: "👨‍👩‍👧",
        participantRoles: ["Organizer", "Family", "Guest"],
        participantSubtitle: "Invite family members into this household",
        heroGradientColors: [Color(hex: "#F59E0B"), Color(hex: "#78350F")],
        pulseHeroGradientColors: [Color(hex: "#FBBF24"), Color(hex: "#B45309").opacity(0.9)],
        statGradients: [
            [Color(hex: "#B45309"), Color(hex: "#92400E")],
            [Color(hex: "#F59E0B"), Color(hex: "#D97706")],
            [Color(hex: "#7C3AED"), Color(hex: "#5B21B6")],
            [Color(hex: "#FBBF24"), Color(hex: "#B45309")],
        ],
        includesContribution: false,
        quickChips: [
            ("👤", "Resident", .resident),
            ("💳", "Expense", .expense),
            ("✅", "Task", .task),
            ("📷", "Memory", .memory),
        ]
    )

    static let customLiving = LivingActiveTheme(
        bg: Color(hex: "#14121B"),
        // Figma 629:15586 hub — emerald #10B981
        accent: Color(hex: "#10B981"),
        accentSolid: Color(hex: "#10B981"),
        accentLight: Color(hex: "#34D399"),
        accentSoft: Color(hex: "#10B981").opacity(0.2),
        text: Color(hex: "#E5E2E1"),
        secondary: Color(hex: "#9CA3AF"),
        muted: Color(hex: "#9CA3AF"),
        card: Color(hex: "#1C1926"),
        border: Color(hex: "#2A2538"),
        darkText: Color(hex: "#14121B"),
        peachChip: Color(hex: "#FBBF24"),
        tealChip: Color(hex: "#14B8A6"),
        purpleChip: Color(hex: "#A855F7"),
        typeLabel: "Custom Living",
        pulseTitle: "Living Pulse",
        contributionsTitle: "Member Contributions",
        budgetTitle: "Property Budget",
        financeTitle: "Property Hub",
        hubHeroAssetName: "CustomLivingHubHero",
        heroEmoji: "✨",
        participantRoles: ["Organizer", "Resident", "Member"],
        participantSubtitle: "Invite people into this custom living space",
        heroGradientColors: [Color(hex: "#10B981"), Color(hex: "#0F766E")],
        pulseHeroGradientColors: [Color(hex: "#34D399"), Color(hex: "#047857").opacity(0.9)],
        statGradients: [
            [Color(hex: "#047857"), Color(hex: "#065F46")],
            [Color(hex: "#10B981"), Color(hex: "#059669")],
            [Color(hex: "#7C3AED"), Color(hex: "#5B21B6")],
            [Color(hex: "#2DD4BF"), Color(hex: "#0F766E")],
        ],
        includesContribution: false,
        quickChips: [
            ("👤", "Resident", .resident),
            ("💳", "Expense", .expense),
            ("✅", "Task", .task),
            ("📷", "Memory", .memory),
        ]
    )

    static func forFamily(_ family: GroupExperienceFamily) -> LivingActiveTheme {
        switch family {
        case .coLiving: return .coLiving
        case .familyHousehold: return .familyHousehold
        case .customLiving: return .customLiving
        default: return .flatmates
        }
    }
}

struct LivingSectionCard<Content: View, Trailing: View>: View {
    let theme: LivingActiveTheme
    let title: String
    @ViewBuilder var trailing: () -> Trailing
    @ViewBuilder var content: () -> Content

    init(
        theme: LivingActiveTheme,
        title: String,
        @ViewBuilder trailing: @escaping () -> Trailing = { EmptyView() },
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.theme = theme
        self.title = title
        self.trailing = trailing
        self.content = content
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text(title)
                    .font(.plusJakarta(size: 18, weight: .heavy))
                    .foregroundStyle(theme.text)
                Spacer()
                trailing()
            }
            content()
        }
        .padding(20)
        .background(theme.card)
        .clipShape(RoundedRectangle(cornerRadius: theme.sectionRadius))
        .overlay(RoundedRectangle(cornerRadius: theme.sectionRadius).stroke(theme.border))
    }
}

struct LivingEmptyBlock: View {
    let theme: LivingActiveTheme
    let message: String
    let detail: String

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(message)
                .font(.plusJakarta(size: 13, weight: .semibold))
                .foregroundStyle(theme.text)
            Text(detail)
                .font(.plusJakarta(size: 12))
                .foregroundStyle(theme.secondary)
        }
    }
}

struct LivingStatCard: View {
    let label: String
    let value: String
    let colors: [Color]
    var icon: String? = nil

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            if let icon {
                Image(systemName: icon)
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundStyle(Color.white.opacity(0.9))
            }
            Text(label)
                .font(.plusJakarta(size: 10, weight: .semibold))
                .foregroundStyle(Color.white.opacity(0.95))
            Text(value)
                .font(.plusJakarta(size: 22, weight: .bold))
                .foregroundStyle(Color.white.opacity(0.95))
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(16)
        .background(LinearGradient(colors: colors, startPoint: .leading, endPoint: .trailing))
        .clipShape(RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(Color.white.opacity(0.1)))
    }
}

struct LivingEmojiChip: View {
    let theme: LivingActiveTheme
    let emoji: String
    let label: String
    var enabled: Bool = false
    var action: () -> Void = {}

    var body: some View {
        Button(action: action) {
            VStack(spacing: 6) {
                Text(emoji)
                    .font(.system(size: 22))
                    .frame(width: 56, height: 56)
                    .background(theme.card)
                    .clipShape(RoundedRectangle(cornerRadius: 16))
                    .overlay(RoundedRectangle(cornerRadius: 16).stroke(theme.border))
                Text(label)
                    .font(.plusJakarta(size: 11, weight: .semibold))
                    .foregroundStyle(theme.text)
            }
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
        .opacity(enabled ? 1 : 0.85)
    }
}

struct LivingCrewRow: View {
    let theme: LivingActiveTheme
    let name: String
    let amountLabel: String
    let percent: Int
    var featured: Bool = false

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            HStack {
                HStack(spacing: 8) {
                    Text(String(name.prefix(1)).uppercased())
                        .font(.plusJakarta(size: 14, weight: .bold))
                        .foregroundStyle(theme.accentLight)
                        .frame(width: 40, height: 40)
                        .background(theme.accentSoft)
                        .clipShape(Circle())
                    VStack(alignment: .leading, spacing: 2) {
                        HStack(spacing: 4) {
                            Text(name)
                                .font(.plusJakarta(size: 13, weight: .semibold))
                                .foregroundStyle(theme.text)
                            if featured {
                                Text("★").foregroundStyle(Color(hex: "#FBBF24")).font(.system(size: 12))
                            }
                        }
                        Text(featured ? "Top contributor" : "Contributor")
                            .font(.plusJakarta(size: 11))
                            .foregroundStyle(theme.muted)
                    }
                }
                Spacer()
                Text(amountLabel)
                    .font(.plusJakarta(size: 13, weight: .bold))
                    .foregroundStyle(theme.text)
            }
            GeometryReader { geo in
                ZStack(alignment: .leading) {
                    Capsule().fill(theme.accentSoft)
                    Capsule()
                        .fill(theme.accent)
                        .frame(width: geo.size.width * CGFloat(min(max(percent, 0), 100)) / 100)
                }
            }
            .frame(height: 6)
        }
    }
}
